### PR TITLE
#4516 New-FormatPs1Xml cmdlet.

### PR DESCRIPTION
**Pull request by vladimir-shcherbakov**
## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#making-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by vladimir-shcherbakov** [comment]
#4516

File: tools/RepoTasks/RepoTasks.Cmdlets/packages.config [hunk 1]
```diff
@@ -0,0 +1,4 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<packages>
+  <package id="System.Management.Automation" version="6.1.7601.17515" targetFramework="net452" />
```
Comment (markcowl) [inline tools/RepoTasks/RepoTasks.Cmdlets/packages.config]: You shouldn't need this the build system should be able to find system.management.automation from your reference assemblies

File: tools/RepoTasks/RepoTasks.RemoteWorker/AppDomainWorker.cs [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+                        var attributes = type.GetCustomAttributes(
+                            typeof(CmdletAttribute),
+                            false);
+
+                        if (attributes.Length == 0) return false;
+                        var cmdletAttribute = (CmdletAttribute)attributes.First();
+                        if (cmdletAttribute == null) return false;
+                        if (!(cmdlets?.Length > 0)) return true;
+                        var attrCmdlet = $"{cmdletAttribute.VerbName}-{cmdletAttribute.NounName}";
+                        //Console.WriteLine("=== attrCmdlet: " + attrCmdlet);
```
Comment (markcowl) [inline tools/RepoTasks/RepoTasks.RemoteWorker/AppDomainWorker.cs]: do we need a logging mechanism?

File: tools/RepoTasks/RepoTasks.RemoteWorker/AppDomainWorker.cs [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+
+        private readonly ISet<string> _usedTypes = new HashSet<string>();
+
+        internal Configuration BuildXmlConfiguration(IEnumerable<Type> cmdletTypList)
+        {
+            var viewList = new List<View>();
+
+            foreach (var cmdletType in cmdletTypList)
+            {
+                Console.WriteLine($@"	cmdlet: {cmdletType.Name}");
```
Comment (markcowl) [inline tools/RepoTasks/RepoTasks.RemoteWorker/AppDomainWorker.cs]: same question

File: tools/RepoTasks/RepoTasks.RemoteWorker/AppDomainWorker.cs [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+
+                var attributes = cmdletType.GetCustomAttributes(
+                    typeof(OutputTypeAttribute),
+                    false);
+
+                if (attributes.Length == 0)
+                {
+                    Console.WriteLine($@"		no OutputTypeAttribute in cmdletType {cmdletType.Name}");
```
Comment (markcowl) [inline tools/RepoTasks/RepoTasks.RemoteWorker/AppDomainWorker.cs]: same comment

File: tools/RepoTasks/RepoTasks.RemoteWorker/AppDomainWorker.cs [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+            IList<TableColumnItem> tableColumnItems, 
+            IList<ListItem> listItems) 
+            where T: MemberInfo
+        {
+
+            var memeber = typeof(T) == typeof(PropertyInfo)
+                ? "property"
+                : "field";
+
+            //var memeber = "memeber";
```
Comment (markcowl) [inline tools/RepoTasks/RepoTasks.RemoteWorker/AppDomainWorker.cs]: remove

File: tools/RepoTasks/RepoTasks.RemoteWorker/AppDomainWorker.cs [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+
+            foreach (var memberInfo in memberInfoListFiltered)
+            {
+                var ps1XmlAttribute = _onlyMarkedProperties
+                    ? (Ps1XmlAttribute)Attribute.GetCustomAttribute(memberInfo, typeof(Ps1XmlAttribute))
+                    : null;
+
+                if (_onlyMarkedProperties && ps1XmlAttribute == null) throw new InvalidCastException("Ps1XmlAttribute");
+
+                Console.WriteLine($@"			{memeber}: {memberInfo.Name}");
```
Comment (markcowl) [inline tools/RepoTasks/RepoTasks.RemoteWorker/AppDomainWorker.cs]: logging?

File: tools/RepoTasks/RepoTasks.RemoteWorker/packages.config [hunk 7]
```diff
@@ -0,0 +1,4 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<packages>
+  <package id="System.Management.Automation" version="6.1.7601.17515" targetFramework="net452" />
```
Comment (markcowl) [inline tools/RepoTasks/RepoTasks.RemoteWorker/packages.config]: same comment

File: Azure.PowerShell.Netcore.sln [hunk 8]
```diff
@@ -53,6 +53,8 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Common.Strategies.Netcore",
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Commands.Common.Tests", "src\Common\Commands.Common.Tests\Commands.Common.Tests.csproj", "{B0EF35E9-2D7A-4AAE-8A1A-728B6CC8524B}"
 EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "RepoTasks.Attributes", "tools\RepoTasks\RepoTasks.Attributes\RepoTasks.Attributes.csproj", "{8FA4A3FF-CB2C-48D9-9B44-A6E01C75FC2A}"
```
Comment (markcowl) [inline Azure.PowerShell.Netcore.sln]: I don't think we want to have this in its own assembly.  I think this should go in a common assembly, likely commands.common

**Comment by vladimir-shcherbakov** [comment]
New PR opened instead:
https://github.com/Azure/azure-powershell/pull/5666

**Pull request closed without merging**